tools/qc/fastq_utils/fastq_pre_barcodes.xml

```diff
@@ -0,0 +1,227 @@
+        <param name="interleaved" label="Interleaved Data" argument="--interleaved" type="text" optional="true" help="Interleaved data, in this format: (read1|read2|index1|index2|index3),(read1|read2|index1|index2|index3)"/>
+        <param name="umi_read" label="UMI read" argument="--umi_read" type="text" optional="true" help="File in which UMI read can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param name="umi_offset" label="UMI offset" argument="--umi_offset" type="integer" optional="true" help="Offset (integer)"/>
+        <param name="umi_size" label="UMI Size" argument="--umi_size" type="integer" optional="true" help="Number of bases after the offset"/>
+        <param name="cell_read" label="Cell Read" argument="--cell_read" type="text" optional="true" help="File in which Cell can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param name="cell_offset" label="Cell Offset" argument="--cell_offset" type="integer" optional="true" help="Offset"/>
+        <param name="cell_size" label="Cell Size" argument="--cell_size" type="integer" optional="true" help="Number of bases after the offset"/>
+        <param name="sample_read" label="Sample Read" argument="--sample_read" type="text" optional="true" help="File in which sample barcode can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param name="sample_offset" label="Sample Offset" argument="--sample_offset" type="integer" optional="true" help="Offset"/>
+        <param name="sample_size" label="Sample Size" argument="--sample_size" type="integer" optional="true" help="Number of bases after the offset"/>
+        <param name="read1_offset" label="read1 Offset" argument="--read1_offset" type="integer" optional="true" help="None"/>
+        <param name="read1_size" label="read1 Size" argument="--read1_size" type="integer" optional="true" help="None"/>
+        <param name="read2_offset" label="read2 Offset" argument="--read2_offset" type="integer" optional="true" help="None"/>
+        <param name="read2_size" label="read2 Size" argument="--read2_size" type="integer" optional="true" help="None"/>
+        <param name="use_10x" label="Use 10x tags" argument="--10x" type="text" optional="true" help="Use 10X UMI tags (UB and UY) instead of the default tags defined in the SAM specification"/>
+        <param name="sam" label="SAMM" argument="--sam" type="text" optional="true" help="No documentation"/>
+        <param name="x" label="X" argument="-X" type="text" optional="true" help="No documentation"/>
+    </inputs>
+    <outputs>
+        <data label="${tool.name} on ${on_string}: Output file 1" name="outfile1" format="fastqsanger" />
```

Comment:
If the tools is directly sending out fastq.gz and the next step uses fastq.gz, then please set the output formats as well to fastqsanger.gz

Reply:
We can later add some conditionality on this, but not needed for now I guess.

Reply:
Done in 36546ac0f9871cb7563c8487ca7c698ba3df9cf0